libtock/util/streaming_process_slice.h

```diff
@@ -0,0 +1,85 @@
+#include "../tock.h"
+
+typedef struct {
+  uint8_t* app_buffer_ptr;
+  size_t app_buffer_size;
+  uint32_t driver;
+  uint32_t allow;
+} streaming_process_slice_state_t;
+
```

Comment:
The header could provide a #define that specifies the expected Header size, so users can rely on that for their buffer-size math.